amd_openvx_extensions/amd_nn/src/batch_normalization_layer.cpp

```diff
@@ -25,15 +25,28 @@ THE SOFTWARE.
 struct BatchNormLayerLocalData {
     NeuralNetworkCommonHandle * handle;
     miopenTensorDescriptor_t input_desc;
+#if ENABLE_OPENCL
     cl_mem input_mem;
+#elif ENABLE_HIP
+    vx_uint8 *input_mem;
+#endif
     miopenTensorDescriptor_t output_desc;
     miopenDataType_t data_type;          // data_type for the kernel
+#if ENABLE_OPENCL
     cl_mem output_mem;
     cl_mem workspace;
+#elif ENABLE_HIP
```

Comment:
if we keep this (void*), we can use the same variables for both HIP and OpenCL. Only cast it where it is used

Reply:
Changing the type of the variable to void* would require changing the call for OCL APIs as well which is not related to this PR. I would suggest keeping the current changes. We can make that change in a separate PR.

Reply:
If you want to keep this format, I would suggest keep all members in the structure within a single #if, #else.

Reply:
switched to void* for LocalData.